lib/models/ReplicationConfiguration.js

```diff
@@ -267,32 +274,35 @@ class ReplicationConfiguration {
     }
 
     /**
-     * Check that the `StorageClass` is a valid class
-     * @param {string} storageClass - The storage class to validate
-     * @return {boolean} `true` if valid, otherwise `false`
-     */
-    _isValidStorageClass(storageClass) {
-        if (!this._config) {
-            return validStorageClasses.includes(storageClass);
-        }
-
-        const replicationEndpoints = this._config.replicationEndpoints
-            .map(endpoint => endpoint.site);
-        return replicationEndpoints.includes(storageClass) ||
-            validStorageClasses.includes(storageClass);
-    }
-
-    /**
      * Check that the `StorageClass` property is valid
      * @param {object} destination - The destination object from this._parsedXML
      * @return {undefined}
      */
     _parseStorageClass(destination) {
         const storageClass = destination.StorageClass &&
             destination.StorageClass[0];
-        if (!this._isValidStorageClass(storageClass)) {
+        if (!this._config) {
+            return validStorageClasses.includes(storageClass);
+        }
+        const replicationEndpoints = this._config.replicationEndpoints
+            .map(endpoint => endpoint.site);
+        const locationConstraints =
+            Object.keys(this._config.locationConstraints);
+        if (locationConstraints.includes(storageClass)) {
+            if (this._prevStorageClass !== null &&
+                this._prevStorageClass !== storageClass) {
+                return errors.InvalidRequest.customizeDescription(
+                    'The storage class must be same for all rules when ' +
+                    'replicating objects to an external location');
+            }
+            this._isExternalLocation = storageClass;
```

Comment:
I guess you want to assign `true` here instead of `storageClass`

Reply:
Thanks! Updated.